src/sysl/core/__main__.py

```diff
@@ -121,8 +124,13 @@ def main():
         except OSError as exc:
             if exc.errno != errno.EEXIST or os.path.isdir(args.output):
                 raise
-
-    args.func(args)
+    try:
+        args.func(args)
+    except Exception as e:
+        if args.trace:
+            raise
+        else:
+            print e
```

Comment:
Since you've added error trapping, I would also add sys.exit(1) here and a sys.exit(0) on the no-error exit path. This makes the tool more usable from scripts that want to detect errors.

Reply:
Great advice, thanks, Done.
I don't think the `sys.exit(0)` is necessary - did you suggest it for symmetry reasons?

Reply:
It's not strictly necessary. I like it because 1) it says "you got here, everything is good", and 2) I lose track of what language does what - e.g. shell will exit with the error code of the last run statement, and 3) "explicit is better than implicit".

But up to you.

Reply:
Totally convinced.